### PR TITLE
Feat/stats analytics page

### PR DESCRIPTION
**Pull request by bcstryker**

### CODE REVIEW

**Comment by bcstryker** [comment]
@janklimo hey, I think this is good to merge now. No errors, all components are working well. There is still some redundancy in the types and maybe some formatting stuff that I'll have to go through and clean up to keep the codebase neat, but no more development really on functionality. The work from here out on this will be organization, styling and refining the codebase.

File: v2/features/stats/jar/MultiLineChart.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  
+  const assetData = data && data.assetData ? data.assetData[timeUnit] : [];
+  const sortedData = assetData
+    ? assetData.sort((a, b) => (a.timestamp > b.timestamp ? 1 : -1))
+    : [];
+  const chartData = sortedData.map(aprAndApyData);
+  return (
+    <LineChart
+      height={500}
+      width={1200}
```
Comment (janklimo) [inline v2/features/stats/jar/MultiLineChart.tsx:38]: The fixed width will not be a viable strategy

File: v2/features/stats/jar/TimeUnitPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React, { FC } from "react";
+
+const button_style = {
+  color: "#26ff91",
+  backgroundColor: "#1f1f1f",
+  marginLeft: "20px",
+  zIndex: 99,
+}
```
Comment (janklimo) [inline v2/features/stats/jar/TimeUnitPanel.tsx:8]: Let's not do this. Create a component, use Tailwind styles.

File: v2/features/stats/platform/ChainTableContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+const formatChainLink = (chainId: string): JSX.Element => (
+  <a href={`./stats/chain?chain=${chainId}`} className={"text-sky-300 hover:underline"}>
+    {chainId.toUpperCase()}
+  </a>
+);
+
+const ChainHead: FC = () => (
+  <thead className={"rounded-t-md"}>
+    <tr className={"w-full border border-slate-600 bg-slate-600"}>
+      <th scope="col" className={"w-1/5 pt-2 pb-2"}>Chain</th>
```
Comment (janklimo) [inline v2/features/stats/platform/ChainTableContainer.tsx:15]: i18n

File: v2/features/stats/platform/common.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import { PickleModelJson } from "picklefinance-core";
+import { ReceiptDataModel, VerboseReceipts } from "./types";
+import { RawChain } from "picklefinance-core/lib/chain/Chains";
+
+const PICKLE_API = `https://f8wgg18t1h.execute-api.us-west-1.amazonaws.com/prod`;
```
Comment (janklimo) [inline v2/features/stats/platform/common.tsx:5]: Let's use the `api.pickle.finance` host.

File: v2/features/stats/platform/common.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  formatPct,
+  formatMoneyChange,
+  formatPctChange,
+  getCurrentReceipts,
+  getLastReceipts,
+  getCurrentReceiptsVerbose,
+  getLastReceiptsVerbose,
+  getCurrentPfCore,
+  getLastPfCore,
+};
```
Comment (janklimo) [inline v2/features/stats/platform/common.tsx:62]: I suggest you don't mix this pattern with individual exports. I'd favor exports in front of any exported entity.

**Review by janklimo — COMMENTED** [review]
Massive lift @bcstryker thanks! Left some ideas for improvement.

File: v2/features/stats/chain/AssetContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
+import { Link, Text } from "@zeit-ui/react";
```
Comment (janklimo) [inline v2/features/stats/chain/AssetContainer.tsx:1]: Let's not use this lib anymore

File: v2/features/stats/chain/AssetContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+import React, { FC } from "react";
+import styles from "../utils/styles.module.css";
```
Comment (janklimo) [inline v2/features/stats/chain/AssetContainer.tsx:3]: And let's not use CSS modules

File: pages/v2/stats/chain.tsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+      getChainData(chain).then((data) => setChainData(data));
+    };
+    getData();
+  }, [chain]);
+
+  return (
+    <div className="block lg:flex mb-8 sm:mb-10">
+      <div className="w-full mb-4 lg:w-1/2 lg:mr-8 lg:mb-0 xl:w-4/5">
+        <ChartContainer chart="tvl" dataSeries={chainData} />
+        <br />
```
[the remark this answers is not in the document]
Comment (bcstryker) [reply pages/v2/stats/chain.tsx:32]: fixed with mb-10

File: pages/v2/stats/index.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+  return (
+    <>
+      <h1 className="font-title font-medium text-2xl sm:text-3xl pt-2">
+        {t("v2.nav.stats")}
+      </h1>
+      <h2 className="font-body font-normal text-foreground-alt-200 text-sm sm:text-base leading-4 sm:leading-6 mt-1">
+        {t("v2.stats.subtitle")}
+      </h2>
+    </>
+  );
```
[the remark this answers is not in the document]
Comment (bcstryker) [reply pages/v2/stats/index.tsx:49]: hey, not sure what you mean

File: public/locales/en/common.json [hunk 10]
```diff
@@ -395,7 +396,40 @@
       "tokenAddress": "Token address"
     },
     "stats": {
-      "subtitle": "Stats provide an overview of total value locked (deposited) in the protocol"
+      "subtitle": "Stats provide an overview of total value locked (deposited) in the protocol",
+      "platform": {
+        "revsChartTitle": "Daily Revenue",
+        "tvlChartTitle": "Total Value Locked"
+      },
+      "chain": {
```
[the remark this answers is not in the document]
Comment (bcstryker) [reply public/locales/en/common.json:404]: fixed

File: v2/features/stats/chain/AssetContainer.tsx (hunk 6 above)
Comment (bcstryker) [reply v2/features/stats/chain/AssetContainer.tsx:1]: deleted this whole file, it was no longer being used

File: v2/features/stats/chain/JarTableContainer.tsx [hunk 11]
```diff
@@ -0,0 +1,78 @@
+import React, { FC } from "react";
+import { formatDollars, formatPercentage } from "v2/utils";
+import { AssetChangeData, AssetData } from "./types";
+
+const formatJarLink = (jarKey: string): JSX.Element => (
+  <a
```
[the remark this answers is not in the document]
Comment (bcstryker) [reply v2/features/stats/chain/JarTableContainer.tsx:6]: done

File: features/Connection/Web3Modal/Web3Modal.tsx [hunk 12]
```diff
@@ -85,6 +85,7 @@ const Web3Modal: FC<Web3ModalProps> = ({ setVisible, ...rest }) => {
         activate(web3connector);
         setVisible(false);
       } else {
+        console.log(connectedChain);
```
[the remark this answers is not in the document]
Comment (bcstryker) [reply features/Connection/Web3Modal/Web3Modal.tsx:88]: removed

File: pages/v2/dill.tsx [hunk 13]
```diff
@@ -2,7 +2,7 @@ import { FC } from "react";
 import { useTranslation } from "next-i18next";
 import { useSelector } from "react-redux";
 
-import type { PickleFinancePage } from "v2/types";
+import type { PickleFinancePage, PlatformData } from "v2/types";
```
[the remark this answers is not in the document]
Comment (bcstryker) [reply pages/v2/dill.tsx:5]: removed

File: v2/features/stats/chain/JarTableContainer.tsx [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+        ))}
+      </tbody>
+    </table>
+  );
+};
+
+export const JarTableContainer: FC<{ assets: AssetData }> = ({ assets }) => {
+  return (
+    <div className="bg-background-light rounded-xl border border-foreground-alt-500 shadow p-4 sm:p-8">
+      <h2 className="font-body font-bold text-xl">Jar Stats</h2>
```
[the remark this answers is not in the document]
Comment (bcstryker) [reply v2/features/stats/chain/JarTableContainer.tsx:73]: done, will continue to make this change wherever I see it is needed

File: v2/features/stats/platform/ChainTableContainer.tsx [hunk 15]
```diff
@@ -0,0 +1,68 @@
+import React, { FC } from "react";
+import { PlatformChainData, PlatformChainCoreData } from "v2/types";
+import { formatDollars } from "v2/utils";
+
+
+const formatChainLink = (chainId: string): JSX.Element => (
+  <a href={`./stats/chain?chain=${chainId}`} className={"text-sky-300 hover:underline"}>
```
[the remark this answers is not in the document]
Comment (bcstryker) [reply v2/features/stats/platform/ChainTableContainer.tsx:7]: done

File: v2/features/stats/platform/common.tsx (hunk 4 above)
Comment (bcstryker) [reply v2/features/stats/platform/common.tsx:5]: deleted file, not needed

File: v2/features/stats/platform/common.tsx (hunk 5 above)
Comment (bcstryker) [reply v2/features/stats/platform/common.tsx:62]: deleted file

File: v2/features/stats/platform/types.ts [hunk 16]
```diff
@@ -0,0 +1,256 @@
...
+import { ChainNetwork, JarHarvestStats } from "picklefinance-core";
+
+export type SortDirection = "asc" | "desc";
+
+export type Sort = {
+  prop: keyof TableData | keyof HarvestSummary | keyof PendingTxTableData | keyof HarvestDelayData;
+  direction: SortDirection;
+};
+
+export interface nameTvlPair {
```
[the remark this answers is not in the document]
Comment (bcstryker) [reply v2/features/stats/platform/types.ts:11]: deleted file, no longer needed

File: v2/features/stats/platform/types.ts [hunk 17]
```diff
@@ -0,0 +1,256 @@
...
+  tvlPctChange?: number;
+  tvlPctChangeFormatted?: string;
+  actions: JSX.Element;
+}
+
+export interface LastHarvestPair {
+  [jarName: string]: number;
+}
+
+export interface LastHarvestJson {
```
[the remark this answers is not in the document]
Comment (bcstryker) [reply v2/features/stats/platform/types.ts:90]: done, deleted entire file

File: pages/v2/stats/chain.tsx [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+import { useTranslation } from "next-i18next";
+import { useRouter } from "next/router";
+import type { PickleFinancePage } from "v2/types";
+import { ChartContainer } from "v2/features/stats/chain/ChartContainer";
+import { ChainData } from "v2/features/stats/chain/types";
+import { JarTableContainer } from "v2/features/stats/chain/JarTableContainer";
+
+const getChainData = async (chain: string): Promise<ChainData> => {
+  return await fetch(
+    `https://api.pickle.finance/prod/protocol/analytics/chain/${chain}/en`,
```
[the remark this answers is not in the document]
Comment (bcstryker) [reply pages/v2/stats/chain.tsx]: added

**Comment by bcstryker** [comment]
closes #355

File: pages/v2/stats/index.tsx [hunk 19]
```diff
@@ -0,0 +1,56 @@
+import { FC, useEffect, useState } from "react";
+import { useTranslation } from "next-i18next";
+import { ChartContainer } from "v2/features/stats/platform/ChartContainer";
+import { ChainTableContainer } from "v2/features/stats/platform/ChainTableContainer";
+import { PlatformData } from "v2/types";
+import type { PickleFinancePage } from "v2/types";
+
+
+
```
[the remark this answers is not in the document]
Comment (bcstryker) [reply pages/v2/stats/index.tsx]: done

File: v2/features/stats/chain/TvlChart.tsx [hunk 20]
```diff
@@ -0,0 +1,79 @@
...
+      <Tooltip 
+        cursor={false}
+        contentStyle={{backgroundColor: "black", color: "#26ff91"}}
+        labelFormatter={(label) => new Date(label).toLocaleDateString() + ' ' + new Date(label).toLocaleTimeString()} 
+        formatter={(value: number) => formatDollars(value)} 
+      />
+      <Line
+        type="monotone"
+        dataKey={"value"}
+        stroke="#53ffe2"
```
[the remark this answers is not in the document]
Comment (bcstryker) [reply v2/features/stats/chain/TvlChart.tsx]: done

File: v2/features/stats/jar/MultiLineChart.tsx (hunk 1 above)
Comment (bcstryker) [reply v2/features/stats/jar/MultiLineChart.tsx]: fixed

**Pull request merged**